Assets/HoloToolkit/Input/Scripts/Utilities/MicrophoneHelper.cs

```diff
@@ -0,0 +1,157 @@
+        /// per-app basis. Along with declaring the microphone DeviceCapability in the package manifest,
+        /// this method tests the privacy setting for microphone access for this application.
+        /// Note that this only checks the Settings->Privacy->Microphone setting, it does not handle
+        /// the Cortana/Dictation privacy check, however (Under Settings->Privacy->Speech, Inking and Typing).
+        /// 
+        /// Developers should ideally perform a check like this every time their app gains focus, in order to 
+        /// check if the user has changed the setting while the app was suspended or not in focus.
+        /// </summary>
+        /// <returns>
+        /// Returns a MicrophoneStatus value indicating the specific issue if microphone cannot be used.
+        /// If everything went well return value is MicrophoneStatus.MicrophoneReady
+        /// </returns>
+        public async static Task<MicrophoneStatus> GetMicrophoneStatus()
+        {
+            // Request access to the microphone only, to limit the number of capabilities we need
+            // to request in the package manifest.
+            MediaCaptureInitializationSettings settings = new MediaCaptureInitializationSettings();
+            settings.StreamingCaptureMode = StreamingCaptureMode.Audio;
+            settings.MediaCategory = MediaCategory.Speech;
+            MediaCapture capture = new MediaCapture();
+            try
+            {
+                // If this is the first time we run the app, this will show the
+                // microphone permissions dialog, where the user is asked if they
+                // allow the app to use the microphone. If the dialog is dismissed, or not
+                // accepted, the app will not ask for permissions again. The user will need
+                // to go to Settings->Privacy->Microphone Settings to allow the app to
+                // use the microphone
+                await capture.InitializeAsync(settings);
+            }
+            catch (TypeLoadException)
+            {
+                // On SKUs without media player (eg, the N SKUs), we may not have access to the Windows.Media.Capture
+                // namespace unless the media player pack is installed.
+                return MicrophoneStatus.MediaComponentsMissing;
+            }
+            catch (UnauthorizedAccessException)
+            {
+                // The user has turned off access to the microphone. If this occurs, we should show an error, or disable
+                // functionality within the app to ensure that further exceptions aren't generated when 
+                // recognition is attempted.
+                return MicrophoneStatus.MicrophoneUseNotAuthorized;
+            }
+            catch (Exception exception)
+            {
+                // This can be replicated by using remote desktop to a system, but not redirecting the microphone input.
```

Comment:
Mixed Reality won't work in remote desktop or VM contexts, so you can probably edit down this comment a little to avoid confusion (still possible to disable all of the audio endpoints though)

Reply:
That's a good point, would this also include Unity's holographic remoting?

Reply:
Holographic remoting is still playing in the editor, where this script would be no-op. I will remove the remote desktop references from the comments.